src/Implement/RequestHandler.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "RequestHandler.h"
+
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <string>
+
+std::unordered_map<std::string, APIHandler *> RequestHandler::apis;
+
+RequestHandler::RequestHandler() { this->read_size = 4096; }
+
+void RequestHandler::SetFd(const int &file_descriptor) {
+  this->client_fd = file_descriptor;
+}
+
+int RequestHandler::ReadHeader(std::string &header,
+                               std::string &msg_over_header) {
+  char input_buffer[this->read_size];
+  const std::string find_key = "Content-Length:";
+  const std::string end_of_header = "\r\n\r\n";
+  // keep read buffer until no request message
+  while (true) {
+    int msg_read = recv(this->client_fd, input_buffer, sizeof(input_buffer), 0);
+    // means end of request
+    if (msg_read <= 0) {
+      return -1;
+    }
+    // add read buffer to header
+    header += input_buffer;
+    // find \r\n\r\n to get end of header
+    size_t start_cut_pos = header.find(find_key, 0) + find_key.length() + 1;
+    size_t end_cut_pos = header.find(end_of_header, 0);
+    if (end_cut_pos != std::string::npos) {
+      try {
+        // convert content-length string to int
+        int client_content_length = std::stoi(
+            header.substr(start_cut_pos, end_cut_pos - start_cut_pos));
+        // it may occur that reading too many info from buffer, need to move the
+        // body part out of header message
+        msg_over_header = header.substr(
```

Comment:
You can extract `end_cut_pos + end_of_header.length()` out as `header_size` to simplify the expression on Line 46-49

Reply:
Done